internal/pkg/deploy/cloudformation/project.go

```diff
@@ -0,0 +1,269 @@
+)
+
+// CreateProjectResources sets up everything required for our project-wide resources.
+// These resources include things that are regional, rather than scoped to a particular
+// environment, such as ECR Repos, CodePipeline KMS keys & S3 buckets.
+// We deploy project resources through StackSets - that way we can have one
+// template that we update and all regional stacks are updated.
+func (cf CloudFormation) CreateProjectResources(project *archer.Project) error {
+	projectConfig := stack.NewProjectStackConfig(project, cf.box)
+
+	// First deploy the project roles needed by StackSets. These roles
+	// allow the stack set to set up our regional stacks.
+	if err := cf.deploy(projectConfig); err == nil {
+		_, err := cf.waitForStackCreation(projectConfig)
+		if err != nil {
+			return err
+		}
+	} else {
+		// If the stack already exists - we can move on
+		// to creating the StackSet.
+		var alreadyExists *ErrStackAlreadyExists
+		if !errors.As(err, &alreadyExists) {
+			return err
+		}
+	}
+
+	blankProjectTemplate, err := projectConfig.ResourceTemplate(&stack.ProjectResourcesConfig{
+		Project: projectConfig.Project.Name,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	_, err = cf.client.CreateStackSet(&cloudformation.CreateStackSetInput{
+		Description:           aws.String(projectConfig.StackSetDescription()),
+		StackSetName:          aws.String(projectConfig.StackSetName()),
+		TemplateBody:          aws.String(blankProjectTemplate),
+		ExecutionRoleName:     aws.String(projectConfig.StackSetExecutionRoleName()),
+		AdministrationRoleARN: aws.String(projectConfig.StackSetAdminRoleARN()),
+		Tags:                  projectConfig.Tags(),
+	})
+
+	if err != nil && !stackSetExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+// AddAppToProject attempts to add new App specific resources to the Project resource stack.
+// Currently, this means that we'll set up an ECR repo with a policy for all envs to be able
+// to pull from it.
+func (cf CloudFormation) AddAppToProject(project *archer.Project, newApp *archer.Application) error {
+	projectConfig := stack.NewProjectStackConfig(project, cf.box)
+	previouslyDeployedConfig, err := cf.getLastDeployedProjectConfig(projectConfig)
+	if err != nil {
+		return fmt.Errorf("adding %s app resources to project %s: %w", newApp.Name, project.Name, err)
+	}
+
+	// We'll generate a new list of Accounts to add to our project
+	// infrastructure by appending the environment's account if it
+	// doesn't already exist.
+	appList := []string{}
+	shouldAddNewApp := true
+	for _, app := range previouslyDeployedConfig.Apps {
+		appList = append(appList, app)
+		if app == newApp.Name {
+			shouldAddNewApp = false
+		}
+	}
+
+	if !shouldAddNewApp {
+		return nil
+	}
+
+	appList = append(appList, newApp.Name)
+
+	newDeploymentConfig := stack.ProjectResourcesConfig{
+		Version:  previouslyDeployedConfig.Version + 1,
+		Apps:     appList,
+		Accounts: previouslyDeployedConfig.Accounts,
```

Comment:
Does the `Accounts` value potentially need to be updated to include the `newApp` `Account`?

Reply:
I think you can disregard, mixed this up with the `AddEnvToProject` flow.